src/OpenFeature.Contrib.Providers.Flagd/Resolver/InProcess/JsonEvaluator.cs

```diff
@@ -2,32 +2,35 @@
 using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using System.Text.Json.Serialization;
 using System.Text.RegularExpressions;
-using JsonLogic.Net;
-using Newtonsoft.Json;
-using Newtonsoft.Json.Linq;
+using Json.Logic;
+using Json.More;
 using OpenFeature.Constant;
 using OpenFeature.Contrib.Providers.Flagd.Resolver.InProcess.CustomEvaluators;
 using OpenFeature.Error;
 using OpenFeature.Model;
+using EvaluationContext = OpenFeature.Model.EvaluationContext;
 
 namespace OpenFeature.Contrib.Providers.Flagd.Resolver.InProcess
 {
     internal class FlagConfiguration
     {
-        [JsonProperty("state")] internal string State { get; set; }
-        [JsonProperty("defaultVariant")] internal string DefaultVariant { get; set; }
-        [JsonProperty("variants")] internal Dictionary<string, object> Variants { get; set; }
-        [JsonProperty("targeting")] internal object Targeting { get; set; }
-        [JsonProperty("source")] internal string Source { get; set; }
-        [JsonProperty("metadata")] internal Dictionary<string, object> Metadata { get; set; }
+        [JsonPropertyName("state")] public string State { get; set; }
+        [JsonPropertyName("defaultVariant")] public string DefaultVariant { get; set; }
+        [JsonPropertyName("variants")] public Dictionary<string, JsonElement> Variants { get; set; }
+        [JsonPropertyName("targeting")] public object Targeting { get; set; }
+        [JsonPropertyName("source")] public string Source { get; set; }
+        [JsonPropertyName("metadata")] public Dictionary<string, JsonElement> Metadata { get; set; }
     }
 
     internal class FlagSyncData
     {
-        [JsonProperty("flags")] internal Dictionary<string, FlagConfiguration> Flags { get; set; }
-        [JsonProperty("$evaluators")] internal Dictionary<string, object> Evaluators { get; set; }
-        [JsonProperty("metadata")] internal Dictionary<string, object> Metadata { get; set; }
+        [JsonPropertyName("flags")] public Dictionary<string, FlagConfiguration> Flags { get; set; }
+        [JsonPropertyName("$evaluators")] public Dictionary<string, object> Evaluators { get; set; }
+        [JsonPropertyName("metadata")] public Dictionary<string, JsonElement> Metadata { get; set; }
```

Comment:
All these properties became public. Any reason?

Reply:
Reverted

Reply:
It appear that with the new JSON serialization, they need to be public. Maybe there's some workaround, but the tests all fail if these are internal.

Reply:
I am fine with keeping them public. 👍